boards/arm/mimxrt1020_evk/mimxrt1020_evk.dts

```diff
@@ -79,13 +81,26 @@ arduino_serial: &lpuart2 {};
 &flexspi {
 	reg = <0x402a8000 0x4000>, <0x60000000 0x800000>;
 	is25wp064: is25wp064@0 {
-		compatible = "issi,is25wp064", "jedec,spi-nor";
+		compatible = "nxp,qspi-nor","jedec,spi-nor";
 		size = <67108864>;
 		label = "IS25WP064";
 		reg = <0>;
-		spi-max-frequency = <133000000>;
+		sck-frequency = <133000000>;
 		status = "okay";
 		jedec-id = [9d 70 17];
+		erase-block-size = <4096>;
+		write-block-size = <4>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			storage_partition: partition@7c0000 {
```

Comment:
Out of curiosity: IIRC on the mimxrt1060-evk the QSPI is used to store the program.  Is there other data on QSPI (I think there might be, based on a recovery process I had to do once)?  Is there documentation on what memory might be available for use?

I don't have the MIMXRT1020-EVK, so could only test this on MIMXRT1060-EVK, and that only if it's possible to share the QSPI device between the code and data.  I'm thinking it probably isn't possible, and I can't convert my board to use hyperflash instead if that's a hardware modification.

Reply:
I don't think the blocking call used to write to flash uses DMA internally, but it is a valid point, at least if someone wants to change to a non-blocking call at some point. Maybe I should add a comment. I saw this being handled in some other driver, with a ram buffer of configurable size or something.

Regarding data on QSPI flash. Are you maybe referring to the boot header read by the boot ROM to configure the flash hardware before the program starts to execute? If you overwrite it, you have to put it back somehow for booting to work again. This is particularly tricky with the details around the non-volatile QSPI Enable bit in the external flash chip. I don't think the Zephyr code generats instructions to set it correctly, and relies on it already being set in the flash chip.

I am still having a hard time to figure out if the driver should rely on the correct configuration already being set up by the header, or if it should reconfigure the flash hardware on init according to the devicetree, possibly screwing up XIP in doing so.

Reply:
@dleach02 @MaureenHelm the above might relate to the fix-broken-imx stuff we were doing a while back, or at least you might have some insight as to how/whether mixed use of the device could be made to work.

I don't think I'll try building/testing this driver.